### PR TITLE
Pledge error handling

### PR DESCRIPTION
**Pull request by goatsOnDrugs**
## Description

Error handling so api will return a message and name so we can show different errors on pledge form and pledge dashboard. I'm not sure where we should be translating the message. The modal styles needs review by Professor still.

## Related Ticket

Resolves #<issue-number>

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

File: site/components/pages/Pledge/PledgeForm/index.tsx [hunk 1]
```diff
@@ -124,6 +124,7 @@ export const PledgeForm: FC<Props> = (props) => {
         reset(pledgeFormAdapter(data.pledge));
         setServerError(false);
       } else {
+        console.log("data", data);
```
[the remark this answers is not in the document]
Comment (goatsOnDrugs) [reply site/components/pages/Pledge/PledgeForm/index.tsx]: sorry, removed

File: site/components/pages/Pledge/PledgeDashboard/index.tsx [hunk 2]
```diff
@@ -68,14 +71,24 @@ export const PledgeDashboard: NextPage<Props> = (props) => {
...
         secondaryWalletAddress: address,
         pledge: pledgeFormAdapter(pledge),
         signature,
       });
-    } catch {
-      console.log("uh ohhh");
+      const data = await response.json();
+      if (data.pledge) {
+        setPledge(data.pledge);
+      } else {
```
[the remark this answers is not in the document]
Comment (goatsOnDrugs) [reply site/components/pages/Pledge/PledgeDashboard/index.tsx]: Ok, that makes sense. Good idea. Implemented.

File: site/components/pages/Pledge/lib/firebase.ts [hunk 3]
```diff
@@ -39,7 +39,11 @@ export const getPledgeByAddress = async (address: string): Promise<Pledge> => {
   const pledgeRef = pledgeSnapshot.docs[0];
 
   if (!pledgeRef) {
-    throw new Error("Bad request");
+    const e = new Error(
+      "Something went wrong on our end. Please try again in a few minutes"
```
[the remark this answers is not in the document]
Comment (goatsOnDrugs) [reply site/components/pages/Pledge/lib/firebase.ts]: Copy updated

File: site/components/pages/Pledge/lib/firebase.ts [hunk 4]
```diff
@@ -39,7 +39,11 @@ export const getPledgeByAddress = async (address: string): Promise<Pledge> => {
...
+    );
+    e.name = "BadRequest";
```
[the remark this answers is not in the document]
Comment (goatsOnDrugs) [reply site/components/pages/Pledge/lib/firebase.ts:46]: updated

**Pull request merged**